accelerators/nvidia.go

```diff
@@ -132,27 +131,32 @@ func (nm *nvidiaManager) GetCollector(devicesCgroupPath string) (stats.Collector
 	nc := &nvidiaCollector{}
 
 	if !nm.devicesPresent {
-		return nc, nil
+		return &stats.NoopCollector{}, nil
 	}
 	// Makes sure that we don't call initializeNVML() concurrently and
 	// that we only call initializeNVML() when it's not initialized.
 	nm.Lock()
 	if !nm.nvmlInitialized {
-		initializeNVML(nm)
-	}
-	if !nm.nvmlInitialized || len(nm.nvidiaDevices) == 0 {
-		nm.Unlock()
-		return nc, nil
+		err := initializeNVML(nm)
+		if err != nil {
+			nm.Unlock()
+			klog.Warningf("Unable to initialize NVML: %s", err)
```

Comment:
Since this is called on each container creation, we probably don't want to Warningf here.  E.g. if nvml isn't installed on the system, this is expected to log.   I think if we just return the error, it will be logged at V(4), which is probably what we want